lib/elix/brain/redis_store.ex

```diff
@@ -0,0 +1,65 @@
+defmodule Elix.Brain.RedisStore do
+  @moduledoc """
+  A Redis back-end for the Brain which persists memory to disk.
+  """
+
+  @behaviour Elix.Brain.Store
+  @encoded_binary_prefix "binterm:"
+
+  def start_link do
+    redis_url = System.get_env("REDIS_URL") || "redis://"
+    {:ok, _pid} = Redix.start_link(redis_url, name: :redis)
+  end
+
+  @spec set(String.t, list) :: any
+  def set(key, list) do
+    command!(["DEL", key])
+    Enum.each(list, fn (item) ->
+      command!(["RPUSH", key, encode_if_necsssary(item)])
+    end)
```

Comment:
I ended up wrapping this in a `MULTI` transaction, and also taking advantage of the fact that `RPUSH` can take an arbitrary number of items in order to append them all in a single operation:

```elixir
def set(key, list) do
  transaction fn ->
    command!(["DEL", key])
    items = Enum.map(list, &encode_if_necsssary(&1))
    command!(["RPUSH", key] ++ items)
  end
end
```

https://github.com/stevegrossi/hedwig_brain/blob/master/lib/brain/redis_store.ex#L18